csrc/scheduler/reduction.cpp

```diff
@@ -388,9 +388,6 @@ std::unique_ptr<ReductionParams> innerReductionHeuristic(
   bool pad_bdimx = bdimx > 16 &&
       bdimx * bdimy <
           (int64_t)at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock;
-  // If barely just covering reduction dim, don't pad to the next warp
-  pad_bdimx = pad_bdimx &&
-      bdimx * inner_reduction_unroll_factor != inner_most_dimension_numel;
```

Comment:
the comment looks a bit strange here.

`bdimx * inner_reduction_unroll_factor != inner_most_dimension_numel;` == `barely just covering reduction dim`?

Reply:
nvm. I think the idea here is that we always want to pad to facilitate warp reduction, hence removing this... I'm curious why we choose not to pad earlier when the problem size is small.

Reply:
I don't know why. Seems a history thing, maybe older generations of card, at least on A100 we still gains speedup.